src/macros.jl

```diff
@@ -1005,34 +1005,128 @@ function parsevariable(_error::Function, infoexpr::VariableInfoExpr, lvalue, lsi
     var
 end
 
-# @variable(m, expr, extra...; kwargs...)
-# where `extra` is a list of extra positional arguments and `kwargs` is a list of keyword arguments.
-#
-# It creates a new variable (resp. a container of new variables) belonging to the model `m` using `buildvariable` to create the variable (resp. each variable of the container) and `addvariable` to add it to the model.
-# The following modifications will be made to the arguments before they are passed to `buildvariable`:
-# * The `expr` argument will not be passed but the expression will be parsed to determine the kind of container needed (if one is needed) and
-#   additional information that will alter what is passed with the keywords `lowerbound`, `upperbound`, `basename`, `start`, `binary`, and `integer`.
-# * The `PSD` and `Symmetric` positional arguments in `extra` will not be passed to `buildvariable`. Instead,
-#    * the `Symmetric` argument will check that the container is symmetric and only allocate one variable for each pair of non-diagonal entries.
-#    * the `PSD` argument will do the same as `Symmetric` but in addition it will specify that the variables created belongs to the PSD cone in the `varCones` field of the model.
-#   Moreover, Int and Bin are special keywords that are equivalent to `integer=true` and `binary=true`.
-# * The keyword arguments start, lowerbound, upperbound, binary, and integer category may not be passed as is to
-#   `buildvariable` since they may be altered by the parsing of `expr` and we may need to pass it pointwise if it is a container since
-#   `buildvariable` is called separately for each variable of the container. Moreover they will be passed inside the VariableInfo object.
-# * A custom error function is passed as positional argument to print the full @variable call before the error message.
-# * The keyword argument basename is not passed to `constructconstraint!` but rather to `addvariable`.
-#
-# Examples:
-# * `@variable(m, x >= 0)` is equivalent to `x = addvariable(m, buildvariable(msg -> error("In @variable(m, x >= 0): ", msg), VariableInfo(true, 0, false, NaN, false, NaN, false, NaN, false, false)), "x")`
-# * `@variable(m, x[1:N,1:N], Symmetric, Poly(X))` is equivalent to
-#   ```
-#   x = Matrix{...}(N, N)
-#   for i in 1:N
-#       for j in 1:N
-#           x[i,j] = x[j,i] = addvariable(m, buildvariable(msg -> error("In @variable(m, x[1:N,1:N], Symmetric, Poly(X)): ", msg), VariableInfo(false, NaN, false, NaN, false, NaN, false, NaN, false, false), Poly(X)), "")
-#       end
-#   end
-#   ```
+"""
+    @variable(m; kwargs...)
```

Comment:
Does the macro call work correctly if you put a semicolon before the keyword arguments? If not, using a semicolon here might be confusing.